src/nexusrpc/handler/_core.py

```diff
@@ -0,0 +1,458 @@
+        Awaitable[StartOperationResultSync[Any]],
+        Awaitable[StartOperationResultAsync],
+    ]: ...
+
+    @abstractmethod
+    def fetch_operation_info(
+        self, ctx: FetchOperationInfoContext, token: str
+    ) -> Union[OperationInfo, Awaitable[OperationInfo]]:
+        """Handle a Fetch Operation Info request.
+
+        Args:
+            ctx: The operation context.
+            token: The operation token.
+        """
+        ...
+
+    @abstractmethod
+    def fetch_operation_result(
+        self, ctx: FetchOperationResultContext, token: str
+    ) -> Union[Any, Awaitable[Any]]:
+        """Handle a Fetch Operation Result request.
+
+        Args:
+            ctx: The operation context.
+            token: The operation token.
+        """
+        ...
+
+    @abstractmethod
+    def cancel_operation(
+        self, ctx: CancelOperationContext, token: str
+    ) -> Union[None, Awaitable[None]]:
+        """Handle a Cancel Operation request.
+
+        Args:
+            ctx: The operation context.
+            token: The operation token.
+        """
+        ...
+
+    def _get_service_handler(self, service_name: str) -> ServiceHandler:
+        """Return a service handler, given the service name."""
+        service = self.service_handlers.get(service_name)
+        if service is None:
+            raise HandlerError(
+                f"No handler for service '{service_name}'.",
+                type=HandlerErrorType.NOT_FOUND,
+            )
+        return service
+
+
+class Handler(BaseHandler):
+    """
+    A Nexus handler manages a collection of Nexus service handlers.
+
+    Operation requests are delegated to a :py:class:`ServiceHandler` based on the service
+    name in the operation context.
+    """
+
+    async def start_operation(
+        self,
+        ctx: StartOperationContext,
+        input: LazyValue,
+    ) -> Union[
+        StartOperationResultSync[Any],
+        StartOperationResultAsync,
+    ]:
+        """Handle a Start Operation request.
+
+        Args:
+            ctx: The operation context.
+            input: The input to the operation, as a LazyValue.
+        """
+        service_handler = self._get_service_handler(ctx.service)
+        op_handler = service_handler._get_operation_handler(ctx.operation)
+        op = service_handler.service.operations[ctx.operation]
+        deserialized_input = await input.consume(as_type=op.input_type)
+
+        if is_async_callable(op_handler.start):
+            # TODO(preview): apply middleware stack as composed awaitables
+            return await op_handler.start(ctx, deserialized_input)
+        else:
+            # TODO(preview): apply middleware stack as composed functions
+            if not self.executor:
+                raise RuntimeError(
+                    "Operation start handler method is not an `async def` but "
+                    "no executor was provided to the Handler constructor. "
+                )
```

Comment:
Can we detect this at construction time? Meaning can we check in the constructor that if an `executor` is not present, all calls on all operation handlers are `async def`?

Reply:
Good call! I've done that.

To recap, `executor` if present is `concurrent.futures.Executor`. The two handler classes are differentiated as follows:

- `nexusrpc.handler.Handler`: Its own methods are `async def`. Optionally accepts an `executor`. If no `executor` then requires all op handler methods to be async defs. At request-handling time, inspects whether an op handler is sync or async. If sync, then submits to executor using `asyncio.get_event_loop().run_in_executor(executor, ...)`, passing it the 

- `nexusrpc.syncio.handler.Handler`: Its own methods are `def`. Requires an `executor`, and requires all operation handler methods to be `def`. Submits to executor using `submit` of the executor.